postgres-appliance/configure_spilo.py

```diff
@@ -347,37 +412,78 @@ def get_dcs_config(config, placeholders):
+        rename = prefix + name
+        if rename in placeholders:
+            wale[name] = placeholders[rename]
+
+    if not os.path.exists(wale['WALE_ENV_DIR']):
+        os.makedirs(wale['WALE_ENV_DIR'])
 
     if provider == PROVIDER_AWS:
-        write_file('s3://{WAL_S3_BUCKET}/spilo/{SCOPE}/wal/'.format(**placeholders),
-                   os.path.join(placeholders['WALE_ENV_DIR'], 'WALE_S3_PREFIX'), overwrite)
-        match = re.search(r'.*(eu-\w+-\d+)-.*', placeholders['WAL_S3_BUCKET'])
+        write_file('s3://{WAL_S3_BUCKET}/spilo/{SCOPE}/wal/'.format(**wale),
+                   os.path.join(wale['WALE_ENV_DIR'], 'WALE_S3_PREFIX'), overwrite)
+        match = re.search(r'.*(eu-\w+-\d+)-.*', wale['WAL_S3_BUCKET'])
         if match:
             region = match.group(1)
         else:
             region = placeholders['instance_data']['zone'][:-1]
         write_file('https+path://s3-{}.amazonaws.com:443'.format(region),
-                   os.path.join(placeholders['WALE_ENV_DIR'], 'WALE_S3_ENDPOINT'), overwrite)
+                   os.path.join(wale['WALE_ENV_DIR'], 'WALE_S3_ENDPOINT'), overwrite)
     elif provider == PROVIDER_GOOGLE:
-        write_file('gs://{WAL_GCS_BUCKET}/spilo/{SCOPE}/wal/'.format(**placeholders),
-                   os.path.join(placeholders['WALE_ENV_DIR'], 'WALE_GS_PREFIX'), overwrite)
+        write_file('gs://{WAL_GCS_BUCKET}/spilo/{SCOPE}/wal/'.format(**wale),
+                   os.path.join(wale['WALE_ENV_DIR'], 'WALE_GS_PREFIX'), overwrite)
         if placeholders['GOOGLE_APPLICATION_CREDENTIALS']:
             write_file('{GOOGLE_APPLICATION_CREDENTIALS}'.format(**placeholders),
-                       os.path.join(placeholders['WALE_ENV_DIR'], 'GOOGLE_APPLICATION_CREDENTIALS'), overwrite)
+                       os.path.join(wale['WALE_ENV_DIR'], 'GOOGLE_APPLICATION_CREDENTIALS'), overwrite)
     else:
         return
+
     if not os.path.exists(placeholders['WALE_TMPDIR']):
         os.makedirs(placeholders['WALE_TMPDIR'])
         os.chmod(placeholders['WALE_TMPDIR'], 0o1777)
 
-    write_file(placeholders['WALE_TMPDIR'], os.path.join(placeholders['WALE_ENV_DIR'], 'TMPDIR'), True)
+    write_file(placeholders['WALE_TMPDIR'], os.path.join(wale['WALE_ENV_DIR'], 'TMPDIR'), True)
+
 
+def write_wale_command_environment(placeholders, provider, overwrite):
+    if not placeholders['USE_WALE']:
+        return
+
+    write_wale_environment(placeholders, provider, '', overwrite)
+
+
+def write_wale_clone_environment(placeholders, provider, overwrite):
+    if not placeholders['CLONE_WITH_WALE']:
+        return
+
+    write_wale_environment(placeholders, provider, 'CLONE_', overwrite)
+
+
+def write_bootstrap_configuration(placeholders, provider, overwrite):
+    write_wale_clone_environment(placeholders, provider, overwrite)
+    write_clone_pgpass(placeholders, overwrite)
+
+
+def write_clone_pgpass(placeholders, overwrite):
+    if not placeholders['CLONE_WITH_BASEBACKUP']:
```

Comment:
Move check to the line https://github.com/zalando/spilo/pull/181/files#diff-9ec119124414a288bc2f1edd468dcce5R468
to avoid calling function when it is not necessary.